timeseries/src/autogluon/timeseries/predictor.py

```diff
@@ -1187,3 +1189,112 @@ def select_target(ts_df: TimeSeriesDataFrame) -> TimeSeriesDataFrame:
             quantile_levels=self.quantile_levels,
         )
         return simulation_dict
+
+    def plot(
+        self,
+        data: Union[TimeSeriesDataFrame, pd.DataFrame, str],
+        predictions: Optional[Union[TimeSeriesDataFrame, pd.DataFrame, str]] = None,
+        quantile_levels: Optional[List[float]] = None,
+        item_ids: Optional[List[Union[str, int]]] = None,
+        max_num_item_ids: int = 8,
+        max_history_length: Optional[int] = None,
+        point_forecast_column: Optional[str] = None,
+        matplotlib_rc_params: Optional[dict] = None,
+    ):
+        """Plot historic time series values and the forecasts.
+
+        Parameters
+        ----------
+        data : Union[TimeSeriesDataFrame, pd.DataFrame, str]
+            Observed time series data.
+        predictions : TimeSeriesDataFrame, optional
+            Predictions generated by calling :meth:`~autogluon.timeseries.TimeSeriesPredictor.predict`.
+        quantile_levels : List[float], optional
+            Quantile levels for which to plot the prediction intervals. Defaults to lowest & highest quantile levels
+            available in ``predictions``.
+        item_ids : List[Union[str, int]], optional
+            If provided, the plots will only be generated for time series with these item IDs. By default, selects
+            up to ``max_num_item_ids`` random item IDs from the dataset.
+        max_num_item_ids : int, default = 8
+            At most this many time series will be plotted by the method.
+        max_history_length : int, optional
+            If provided, at most this many time steps will be shown for each time series in ``data``.
+        point_forecast_column : str, optional
+            Name of the column in ``predictions`` that will be plotted as the point forecast. Defaults to ``"0.5"``,
+            if this column is present in ``predictions``, otherwise ``"mean"``.
+        matplotlib_rc_params : dict, optional
+            Dictionary describing the plot style that will be passed to [`matplotlib.pyplot.rc_context`](https://matplotlib.org/stable/api/_as_gen/matplotlib.pyplot.rc_context.html).
+            See [matplotlib documentation](https://matplotlib.org/stable/users/explain/customizing.html#the-default-matplotlibrc-file) for the list of available options.
+        """
+        import matplotlib.pyplot as plt
+
+        data = self._check_and_prepare_data_frame(data)
+        if item_ids is None:
+            item_ids = list(np.random.choice(data.item_ids, size=min(max_num_item_ids, data.num_items), replace=False))
+        else:
+            item_ids = list(item_ids)[:max_num_item_ids]
```

Comment:
So If I give you 9 item_ids you will not plot the 9th... The `item_ids` docstring does not spell that out very well, and I would expect this not to be the case.

Reply:
I have updated the docstrings to communicate this more clearly.

I have two arguments in favor of keeping the current argument name & effect:

1. I think that limiting the max number of plots to `max_num_item_ids=8` by default is a reasonable safeguard in case the user calls
```python
predictor.plot(data, item_ids=data.item_ids)
```
which may generate thousands of plots and kill the jupyter kernel.
2. If `max_num_item_ids` only applies to the case when `item_ids=None`, we'll need to rename to it something like `max_num_random_item_ids` or `max_num_default_item_ids`, which doesn't sound as intuitive as the current name.